components/src/tooltips/DeprecatedTooltip.tsx

```diff
@@ -1,55 +1,48 @@
-// @flow
-
 import * as React from 'react'
 import { Manager, Reference, Popper } from 'react-popper'
 import cx from 'classnames'
 import styles from './tooltips.css'
 
 const DISTANCE_FROM_REFERENCE = 8
 
-type PopperProps = React.ElementProps<typeof Popper>
-
-export type TooltipChildProps<ChildProps: {}> = {|
-  ...$Exact<ChildProps>,
-  ref: React.Ref<*>,
-|}
+type PopperProps = React.ComponentProps<typeof Popper>
 
-export type DeprecatedTooltipProps<ChildProps: {}> = {|
+export interface DeprecatedTooltipProps {
   /** show or hide the tooltip */
-  open?: boolean,
+  open?: boolean
   /** contents of the tooltip */
-  tooltipComponent?: React.Node,
+  tooltipComponent: JSX.Element | unknown
   /** optional portal to place the tooltipComponent inside */
-  portal?: React.ComponentType<*>,
+  portal?: React.ComponentType<any>
   /** <https://github.com/FezVrasta/react-popper#placement> */
-  placement?: $PropertyType<PopperProps, 'placement'>,
+  placement?: PopperProps['placement']
   /** <https://github.com/FezVrasta/react-popper#positionfixed> */
-  positionFixed?: $PropertyType<PopperProps, 'positionFixed'>,
+  positionFixed?: PopperProps['positionFixed']
   /** <https://github.com/FezVrasta/react-popper#modifiers> */
-  modifiers?: $PropertyType<PopperProps, 'modifiers'>,
+  modifiers?: PopperProps['modifiers']
   /** render function for tooltip'd component */
-  children: (props?: TooltipChildProps<ChildProps>) => React.Node,
+  children: (props?: React.PropsWithRef<any>) => JSX.Element | null
```

Comment:
I think `ReactNode` might be better here since it's a render function, not a component

```suggestion
  children: (props?: React.PropsWithRef<any>) => React.ReactNode
```

Reply:
In this case, if I typed this as `React.ReactNode` the Functional Component's return value below wouldn't be valid, because `React.ReactNode` can be undefined.  TS is happy with it this way, and because this component is deprecated anyway, it's probably fine for now.